src/core/Renderer.js

```diff
@@ -125,6 +125,11 @@ export class Renderer {
         this.gl.viewport(0, 0, width, height);
     }
 
+    setScissor(width, height, x = 0, y = 0) {
+        this.enable(this.gl.SCISSOR_TEST);
+        this.gl.scissor(x, y, width, height);
+    }
+
```

Comment:
Not sure if I like enabling scissor test here as users would have to disable it on their own. Might be too much magic.